test_unstructured_ingest/unit/pipeline/reformat/test_chunking.py

```diff
@@ -0,0 +1,147 @@
+from __future__ import annotations
+
+import logging
+import os
+
+import pytest
+from _pytest.logging import LogCaptureFixture
+
+from test_unstructured.unit_utils import (
+    FixtureRequest,
+    Mock,
+    example_doc_path,
+    function_mock,
+    method_mock,
+)
+from unstructured.documents.elements import CompositeElement
+from unstructured.ingest.interfaces import ChunkingConfig, PartitionConfig
+from unstructured.ingest.pipeline.interfaces import PipelineContext
+from unstructured.ingest.pipeline.reformat.chunking import Chunker
+
+ELEMENTS_JSON_FILE = example_doc_path(
+    "test_evaluate_files/unstructured_output/Bank Good Credit Loan.pptx.json"
+)
+
+
+class DescribeChunker:
+    """Unit tests for ingest.pipeline.reformat.chunking.Chunker"""
+
+    # -- Chunker.run() -----------------------------------------------------------------------------
+
+    # -- integration test --
+    def it_creates_json(self, _ingest_docs_map_: Mock, tmpdir: str):
+        chunker = Chunker(
+            chunking_config=ChunkingConfig(chunking_strategy="by_title"),
+            pipeline_context=PipelineContext(work_dir=tmpdir),
+            partition_config=PartitionConfig(),
+        )
+        # -- `Chunker.chunk()` defaults to writing to "{work_dir}/chunked", which is located in
+        # -- "/.cache" of a user's profile.
+        # -- Define `work_dir` add the "/chunked" subdirectory to it:
+        os.makedirs(os.path.join(tmpdir, "chunked"), exist_ok=True)
+
+        filename = chunker.run(ELEMENTS_JSON_FILE)
+
+        head, tail = os.path.split(filename if filename else "")
+        # -- Check that a json file was created in `/chunked` --
+        assert head.endswith("chunked")
+        assert tail.endswith(".json")
+
+    def it_returns_None_and_logs_message_without_chunking_strategy(
+        self, _ingest_docs_map_: Mock, caplog: LogCaptureFixture
+    ):
+        chunker = Chunker(
+            chunking_config=ChunkingConfig(),
+            pipeline_context=PipelineContext(),
+            partition_config=PartitionConfig(),
+        )
+        caplog.set_level(logging.INFO)
+
+        assert chunker.run(ELEMENTS_JSON_FILE) is None
+        assert "chunking_strategy is None, skipping chunking for" in caplog.text
+
+    def it_logs_error_with_invalid_remote_chunking_args(
+        self, _ingest_docs_map_: Mock, caplog: LogCaptureFixture
+    ):
+        chunker = Chunker(
+            chunking_config=ChunkingConfig(chunking_strategy="by_invalid"),
+            pipeline_context=PipelineContext(),
+            partition_config=PartitionConfig(partition_by_api=True),
+        )
+
+        chunker.run(ELEMENTS_JSON_FILE)
+
+        assert "Input should be 'basic', 'by_page', 'by_similarity'" in caplog.text
```

Comment:
Why is this the output and where does it come from? Only the remote chunker can tell what chunking strategies it has registered. Seems like `'by_title'` should be in the list too.

And `Input should be ...` isn't very helpful as an error message goes. Is there more that was cut off? I'd be looking to see something saying "chunking_strategy" to give a hint where the problem is. No reason this string can't be as long as it needs to be to clearly represent the log entry.

Reply:
`unstructured_client.models.errors.httpvalidationerror.HTTPValidationError` raises the error and `by_title` is in the list. I was trying to keep it to one line while still conveying what the logged message entailed.

The full output:
```
'ERROR    unstructured.ingest:chunking.py:83 failed to run chunking on file /Users/johnjennings/src/unstructured/example-docs/test_evaluate_files/unstructured_output/Bank Good Credit Loan.pptx.json, {"detail":[{"loc":["body","chunking_strategy"],"msg":"Input should be \'basic\', \'by_page\', \'by_similarity\' or \'by_title\'","type":"literal_error"}]}
Traceback (most recent call last):
  File "/Users/johnjennings/src/unstructured/unstructured/ingest/pipeline/reformat/chunking.py", line 64, in run
   chunked_elements = self.chunk(elements_json)
  File "/Users/johnjennings/src/unstructured/unstructured/ingest/pipeline/reformat/chunking.py", line 109, in chunk
    return partition_via_api(
  File "/Users/johnjennings/src/unstructured/unstructured/partition/api.py", line 103, in partition_via_api\n    response = sdk.general.partition(req)
  File "/Users/johnjennings/.pyenv/versions/unstructured/lib/python3.10/site-packages/unstructured_client/utils/_human_utils.py", line 86, in wrapper
    return func(*args, **kwargs)
  File "/Users/johnjennings/.pyenv/versions/unstructured/lib/python3.10/site-packages/unstructured_client/utils/_human_split_pdf.py", line 40, in wrapper
    return func(*args, **kwargs)
  File "/Users/johnjennings/.pyenv/versions/unstructured/lib/python3.10/site-packages/unstructured_client/general.py", line 90, in partition
    raise out
unstructured_client.models.errors.httpvalidationerror.HTTPValidationError: {"detail":[{"loc":["body","chunking_strategy"],"msg":"Input should be \'basic\', \'by_page\', \'by_similarity\' or \'by_title\'","type":"literal_error"}]}'
```